doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+# User Management design #
+
+##  1. Table of Content
+
+
+* 1. [Table of Content](#TableofContent)
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+		* 1.5.1. [Define defaults](#Definedefaults)
+		* 1.5.2. [Configurations](#Configurations)
+* 2. [Design](#Design)
+	* 2.1. [Architecture Design](#ArchitectureDesign)
+	* 2.2. [High-Level Design](#High-LevelDesign)
+	* 2.2. [Usercfd Role](#UsercfdRole)
+* 3. [Configuration and management](#Configurationandmanagement)
+		* 3.1. [ConfigDB Tables](#ConfigDBTables)
+		* 3.2. [ConfigDB schemas](#ConfigDBschemas)
+		* 3.3. [CLI/YANG model](#CLIYANGmodel)
+* 4. [Build](#Build)
+		* 4.1. [Compilation](#Compilation)
+		* 4.2. [Feature default](#Featuredefault)
+* 5. [Flows](#Flows)
+		* 5.1. [Init Flow](#InitFlow)
+		* 5.2. [Upgrade Flow](#UpgradeFlow)
+		* 5.3. [User add/delete](#UserAddDelete)
+		* 5.4. [User state](#UserState)
+		* 5.5. [User full-name](#UserFullName)
+		* 5.6. [User role](#UserRole)
+		* 5.7. [User hashed-password](#UserHashedPassword)
+        * 5.8. [User password expiration](#UserPasswordExpiration)
+        * 5.9. [User passwdrd prompt](#UserPasswdrdPrompt)
+* 6. [SAI API](#SAIAPI)
+* 7. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+* 8. [Restrictions/Limitations](#RestrictionsLimitations)
+* 9. [Test Plan](#TestPlan)
+		* 9.1. [Unit Test cases](#UnitTestcases)
+
+
+
+###  1.1. Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 12/2022 | Mohammed Zayadna | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This HLD document described the requirements, architecture and configuration details of User Management feature in switches Sonic OS based.
```

Comment:
Does the scope of this document cover local user accounts only?

I feel that we should try to make sure that this design makes space for, or at least avoids conflicting with, remote authentication servers and the management of remotely authenticated users. 

There was a "AAA Improvements" HLD proposed and merged a few years ago (https://github.com/sonic-net/SONiC/blob/master/doc/aaa/AAA%20Improvements/AAA%20Improvements.md) but the code PRs didn't make it in (https://github.com/sonic-net/sonic-buildimage/pull/5553). 
I'm not saying this needs to be revived necessarily, but it is indicative of the complexity that may be involved.

Reply:
I have reviewed the implementation of remote authentication servers and I think that there is no conflict.
I still need to do some manual tests then I will add a new section for AAA integration and elaborate about it.
Regarding "AAA Improvements" HLD, I wasn't aware of it.
Is there still an intention to merge it?
it is obviously conflicts with this HLD

Reply:
I am seeking to revive the effort to contribute the AAA improvements code to SONiC, as the original contributor is no longer assigned to the SONiC project. It would be ideal if we modified both proposals to be compatible with each other so that we could allow for the eventual contribution of AAA enhancements. 
What are some of the conflicts you see? Is it just that the proposed DB schemas are different?